homeassistant/components/lovelace/__init__.py

```diff
@@ -250,6 +291,85 @@ def delete_card(fname: str, card_id: str, position: int = None) -> None:
         "Card with ID: {} was not found in {}.".format(card_id, fname))
 
 
+def get_view(fname: str, view_id: str, data_format: str = FORMAT_YAML) -> None:
+    """Get view without it's cards."""
+    round_trip = data_format == FORMAT_YAML
+    config = yaml.load_yaml(fname, round_trip)
+    for view in config.get('views', []):
+        if str(view.get('id', '')) != view_id:
+            continue
+        del view['cards']
+        if data_format == FORMAT_YAML:
+            return yaml.object_to_yaml(view)
+        return view
+
+    raise ViewNotFoundError(
+        "View with ID: {} was not found in {}.".format(view_id, fname))
+
+
+def update_view(fname: str, view_id: str, view_config, data_format:
+                str = FORMAT_YAML) -> None:
+    """Update view."""
+    config = yaml.load_yaml(fname, True)
+    for view in config.get('views', []):
+        if str(view.get('id', '')) != view_id:
+            continue
+        if data_format == FORMAT_YAML:
+            view_config = yaml.yaml_to_object(view_config)
+        view_config['cards'] = view.get('cards', [])
+        view.clear()
+        view.update(view_config)
+        yaml.save_yaml(fname, config)
+        return
+
+    raise ViewNotFoundError(
+        "View with ID: {} was not found in {}.".format(view_id, fname))
+
+
+def add_view(fname: str, view_config: str,
+             position: int = None, data_format: str = FORMAT_YAML) -> None:
+    """Add a view."""
+    config = yaml.load_yaml(fname, True)
+    views = config.get('views', [])
+    if data_format == FORMAT_YAML:
+        view_config = yaml.yaml_to_object(view_config)
+    if position is None:
+        views.append(view_config)
+    else:
+        views.insert(position, view_config)
+    yaml.save_yaml(fname, config)
+
+
+def move_view(fname: str, view_id: str, position: int) -> None:
+    """Move a view to a different position."""
+    config = yaml.load_yaml(fname, True)
+    views = config.get('views', [])
+    for view in views:
+        if str(view.get('id', '')) != view_id:
+            continue
+        views.insert(position, views.pop(views.index(view)))
+        yaml.save_yaml(fname, config)
+        return
+
+    raise ViewNotFoundError(
+        "View with ID: {} was not found in {}.".format(view_id, fname))
+
+
+def delete_view(fname: str, view_id: str) -> None:
+    """Delete a view."""
+    config = yaml.load_yaml(fname, True)
+    views = config.get('views', [])
+    for view in views:
+        if str(view.get('id', '')) != view_id:
+            continue
+        views.pop(views.index(view))
+        yaml.save_yaml(fname, config)
```

Comment:
Style-wise, I think a for-loop here is not that nice as the code inside is only executed once. Maybe something like this?

```python
view = next((view for view in views if str(view.get('id', '')) == view_id), None)
if view is None:
  raise ViewNotFoundError(...)
views.pop(views.index(view))
# ...
```

Reply:
Whoops, already merged while writing this 😅 Nevermind

Reply:
I think that preferably, we find the object in the for loop, then break. If nothing found, raise error, then mutate and save.

Reply:
Which btw, would be the for-loop equivalent of `next`.